src/backgroundblurprocessor/BackgroundBlurProcessorBuiltIn.ts

```diff
@@ -0,0 +1,139 @@
+import BackgroundFilterSpec from './BackgroundFilterSpec';
+
+/**
+ * The [[BackgroundBlurProcessorBuiltIn]] uses the browser's built-in capability to apply blurring to
+ * the background image as apposed to [[BackgroundBlurProcessorProvided]] that uses WASM and
+ * TensorFlow Lite to apply the blur.
+ */
+
+/** @internal */
+export default class BackgroundBlurProcessorBuiltIn extends BackgroundBlurProcessorProvided {
+  private blurredImage: ImageData;
+  private blurCanvas: HTMLCanvasElement = document.createElement('canvas') as HTMLCanvasElement;
+  private blurCanvasCtx = this.blurCanvas.getContext('2d');
+
+  /**
+   * A constructor that will apply default values if spec and strength are not provided.
+   * If no spec is provided the selfie segmentation model is used with default paths to CDN for the
+   * worker and wasm files used to process each frame.
+   * @param spec The spec defines the assets that will be used for adding background blur to a frame
+   * @param options How much blur to apply to a frame
+   */
+  constructor(spec?: BackgroundFilterSpec, options?: BackgroundBlurOptions) {
+    super(spec, options);
+
+    this.blurCanvas.width = this.spec.model.input.width;
+    this.blurCanvas.height = this.spec.model.input.height;
+    this.logger.info('BackgroundBlur processor using builtin blur');
+  }
+
+  drawImageWithMask(inputCanvas: HTMLCanvasElement, mask: ImageData): void {
+    // Mask will not be set until the worker has completed handling the predict event. Until the first frame is processed,
+    // the whole frame will be blurred.
+    const blurredImage = this.blurredImage;
+    const { canvasCtx, targetCanvas } = this;
+    const { width, height } = targetCanvas;
+
+    if (!mask || !blurredImage) {
+      canvasCtx.clearRect(0, 0, width, height);
+      return;
+    }
+
+    const scaledCtx = this.scaledCanvas.getContext('2d');
+
+    scaledCtx.putImageData(mask, 0, 0);
+    this.blurCanvasCtx.putImageData(this.blurredImage, 0, 0);
+
+    // draw the mask
+    canvasCtx.save();
+    canvasCtx.clearRect(0, 0, width, height);
+    canvasCtx.drawImage(this.scaledCanvas, 0, 0, width, height);
+
+    // Only overwrite existing pixels.
+    canvasCtx.globalCompositeOperation = 'source-in';
+    // draw image over mask...
+    canvasCtx.drawImage(inputCanvas, 0, 0, width, height);
+
+    // draw under person
+    canvasCtx.globalCompositeOperation = 'destination-over';
+    canvasCtx.drawImage(this.blurCanvas, 0, 0, width, height);
+    canvasCtx.restore();
+  }
+
+  /**
+   * This setter will allow setting the blur strength mid stream. The process method will use this
+   * setting on the next frame that is processed.
+   */
+  setBlurStrength(blurStrength: number): void {
+    super.setBlurStrength(blurStrength);
+
+    if (this.worker) {
+      // live update
+      this.modelInitialized = false;
+
+      this.worker.postMessage({ msg: 'destroy' });
+
+      const model = this.spec.model;
```

Comment:
I see the below code appearing twice in some places:
```
const model = this.spec.model;
      this.worker.postMessage({
        msg: 'loadModel',
        payload: {
          modelUrl: model.path,
          inputHeight: model.input.height,
          inputWidth: model.input.width,
          inputChannels: 4,
          modelRangeMin: model.input.range[0],
          modelRangeMax: model.input.range[1],
          blurPixels: this.blurAmount,
        },
      });
```

In `handleInitialize` as well as in `setBlueStrength`, so should we abstract it out in a private method or is there a reason to keep them separate.